### PR TITLE
Make python a required build dependency

### PR DESCRIPTION
**Pull request by skullydazed**
As described in #5776 this adds python to all the OS install scripts.

## Description

We need to test on the following platforms:

* [ ] FreeBSD
* [ ] Fedora
* [x] Debian/Ubuntu
* [x] Arch/Manjaro
* [ ] Gentoo
* [x] sabayon
* [ ] OpenSUSE/Tumbleweed
* [x] Slackware
* [x] macOS
* [x] msys2
* [x] WSL

## Types of Changes

- [x] Core
- [ ] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by skullydazed** [comment]
If you have one of the affected systems please give this a try to make sure it works. 

If you don't yet have python 3 installed:

1. Merge this PR into your tree
2. Build a keyboard, for example `make clueboard/66/rev3`
3. You should get a message saying `Python 3 is not installed. It will be required by a future version of qmk_firmware`
4. Run `util/qmk_install.sh`
5. Run the make command again, the warning message should not show up

If you already have python3 installed you can skip steps 2, 3, and 4.

**Comment by skullydazed** [comment]
Tagging people who have contributed some of the less used platforms. If you are on this list please take a look at this PR and ensure that your OS still works and installs python3.

@manuel-arguelles @Chris-Johnston @vatho @florensie @zjp @vomindoraan @BlitzKraft @shieldsd

**Comment by BlitzKraft** [comment]
@skullydazed For sabayon, python3 is the default python. Changing the line 109 to `dev-lang/python` will suffice. Recent updates dropped python2. Python2 requires explicit version pinning. 

Also, the colon in the line 109 is not correct. It needs to be a hyphen, for specifying version numbers. For python3.5, the package name is `dev-lang/python-3.5.5-r1`. 

I tested the script, and works with the corrections. Without specifying version, python3.6.6 is installed. With specifying 3.5.5-r1, the corresponding package is installed.

**Comment by manuel-arguelles** [comment]
+1 for Slackware

**Comment by skullydazed** [comment]
@BlitzKraft Thanks for the report. Does my latest commit fix those issues?

**Comment by BlitzKraft** [comment]
Yes, it does fix the issues @skullydazed

**Comment by zjp** [comment]
It might take me a minute to check -- it's right around finals for me and every class I'm in has a project due next week. If no one has checked on Gentoo by Friday, I'll look at it.

**Comment by skullydazed** [comment]
@zjp That's fine. If we end up merging this sooner and there is a problem with gentoo a new PR can be created to fix it up.

**Comment by yanfali** [comment]
Tested it on arch linux updated today. Kernel 5.0.10.

1. installed using ./install_linux.sh
  it re-installed stuff but added arm gcc. No errors
2. compiled AVR keyboard `make keebio/iris:default`
  No errors encountered
3. compiled ARM keyboard `make clueboard/60:default`
  1 error encountered. Needed `make git-submodule`

LGTM

File: Makefile [hunk 1]
```diff
@@ -534,6 +534,8 @@ endef
 %:
 	# Check if we have the CMP tool installed
 	cmp $(ROOT_DIR)/Makefile $(ROOT_DIR)/Makefile >/dev/null 2>&1; if [ $$? -gt 0 ]; then printf "$(MSG_NO_CMP)"; exit 1; fi;
+	# Ensure that python3 is installed. This check can be removed after python is used in more places.
+	if ! python3 --version 1> /dev/null 2>&1; then printf "$(MSG_PYTHON_MISSING)"; fi
```
Comment (vomindoraan) [inline Makefile]:
```suggestion
	if ! python3 --version >/dev/null 2>&1; then printf "%s" "$(MSG_PYTHON_MISSING)"; fi
```
`SC2059: Don't use variables in the printf format string. Use printf "..%s.." "$foo".`
Also change the stdout redirection to be consistent with the rest of the file.

File: message.mk [hunk 2]
```diff
@@ -81,3 +81,7 @@ MSG_FILE_TOO_BIG = $(ERROR_COLOR)The firmware is too large!$(NO_COLOR) $(CURRENT
 MSG_FILE_TOO_SMALL = The firmware is too small! $(CURRENT_SIZE)/$(MAX_SIZE)\n
 MSG_FILE_JUST_RIGHT = The firmware size is fine - $(CURRENT_SIZE)/$(MAX_SIZE) ($(FREE_SIZE) bytes free)\n
 MSG_FILE_NEAR_LIMIT = The firmware size is approaching the maximum - $(CURRENT_SIZE)/$(MAX_SIZE) ($(FREE_SIZE) bytes free)\n
+MSG_PYTHON_MISSING = $(WARN_COLOR)WARNING:$(NO_COLOR)\n \
```
Comment (vomindoraan) [inline message.mk]:
```suggestion
MSG_PYTHON_MISSING = $(WARN_COLOR)WARNING:$(NO_COLOR)\n\
```
I think there's no need for the extra space.

**Comment by zjp** [comment]
Sorry to necro this so late but I was able to get my environment set up and find the time to test some things. Per the [Python](https://wiki.gentoo.org/wiki/Python) page on the wiki, it's very integral to Gentoo, so it's unlikely to not be installed on the system. The default right now is Python 3.6 

I'll say looks good to me, even without dev-lang/python in the install script. If, periodically, pulling the repo breaks something for me, I'll take a look at why and update.

**Pull request merged**